R/xy-datamapping.R

```diff
@@ -29,13 +78,129 @@ XYEDataMapping <- R6::R6Class(
 
     # Example of how to do some other stuff
     initialize = function(x, y,
-                          error = NULL,
-                          errorMin = NULL,
-                          errorMax = NULL) {
-      super$initialize(x, y)
+                              error = NULL,
+                              errorMin = NULL,
+                              errorMax = NULL, ...) {
+      super$initialize(x, y, ...)
       self$error <- error
       self$errorMin <- error %||% errorMin
       self$errorMax <- error %||% errorMax
+    },
+
+    getMapData = function(data, metaData) {
+      x <- data[, self$x]
+      y <- data[, self$y]
+      errorMin <- data[, self$errorMin]
+      errorMax <- data[, self$errorMax]
+
+      color <- getDefaultCaptions(
+        data = data,
+        metaData = metaData,
+        variableList = self$color
+      )
+
+      shape <- getDefaultCaptions(
+        data = data,
+        metaData = metaData,
+        variableList = self$shape
+      )
+
+      size <- getDefaultCaptions(
+        data = data,
+        metaData = metaData,
+        variableList = self$size
+      )
+
+      return(list(
+        "x" = x, "y" = y, "ymin" = y - errorMin, "ymax" = y + errorMax,
+        "color" = color, "shape" = shape, "size" = size
+      ))
+    }
+  )
+)
+
+#' @title XRangeDataMapping
+#' @docType class
+#' @description Abstract class for X Range Mapping
+#' @export
+XRangeDataMapping <- R6::R6Class(
+  "XRangeDataMapping",
+  inherit = XYDataMapping,
+  public = list(
+    yMin = NULL,
+    yMax = NULL,
+
+    # Example of how to do some other stuff
+    initialize = function(x, yMin, yMax) {
+      super$initialize(x, y = NULL, ...)
+      self$yMin <- yMin
+      self$yMax <- yMax
+    },
+
+    getMappedData = function(data, metaData) {

```

Comment:
It is supposed to override the `getMapData` from the base class? beware of naming